### PR TITLE
feat: access keys

### PR DESCRIPTION
**Pull request by raymondkarstensen**
fixes #143 
Implemented keys:
Page down, right arrow -> Next page
Page up, left arrow -> Previous page
Home -> Go to first page
End -> Go to last page
n -> Next hit
p -> Previous hit
plus -> zoom in
minus -> zoom out
0 -> zoom out to home
s -> Open search dialog
c -> Toggle contents dialog
f -> Toggle fullscreen

Letter keys is disabled when search dialog is open.
Arrow keys is disabled when zoomed in. When zoomed in the arrow keys will pan.

### CODE REVIEW

File: integration/e2e/pages/viewer.po.ts [hunk 1]
```diff
@@ -80,6 +89,7 @@ export class ViewerPage {
   async openContentSearchDialog() {
     await element(by.css('#contentSearchDialogButton')).click();
     await utils.waitForElement(element(by.css('.content-search-container')));
+    await this.waitForAnimation();
```
Comment (ronnymikalsen) [inline integration/e2e/pages/viewer.po.ts]: Do we need this? waitForElement has already waited for animation

File: integration/e2e/pages/viewer.po.ts [hunk 2]
```diff
@@ -341,6 +351,42 @@ export class ViewerPage {
...
+    } else if (key === 's') {
+      iKey = Key.chord('s');
+    } else if (key === 'c') {
+      iKey = Key.chord('c');
+    } else if (key === 'Esc') {
+      iKey = Key.ESCAPE;
+    }
+
+    await browser.actions().sendKeys(iKey).perform();
+    return await browser.sleep(await this.getAnimationTime() * 1000);
```
Comment (ronnymikalsen) [inline integration/e2e/pages/viewer.po.ts]: Rename getAnimationTime to getAnimationTimeInSec. And create a getAnimationTimeInMs

File: src/lib/src/content-search-dialog/content-search-dialog.service.ts [hunk 3]
```diff
@@ -8,11 +8,12 @@ import { ContentSearchDialogConfigStrategyFactory } from './content-search-dialo
 import { IiifManifestService } from './../core/iiif-manifest-service/iiif-manifest-service';
 import { MimeResizeService } from './../core/mime-resize-service/mime-resize.service';
 import { Manifest } from './../core/models/manifest';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
 
 @Injectable()
 export class ContentSearchDialogService {
   private _el: ElementRef;
-  private isContentSearchDialogOpen = false;
+  private _isContentSearchDialogOpen: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```
Comment (ronnymikalsen) [inline src/lib/src/content-search-dialog/content-search-dialog.service.ts]: We do not start with _ in private attributes names

File: src/lib/src/content-search-dialog/content-search-dialog.service.ts [hunk 4]
```diff
@@ -21,9 +22,13 @@ export class ContentSearchDialogService {
     private contentSearchDialogConfigStrategyFactory: ContentSearchDialogConfigStrategyFactory,
     private mimeResizeService: MimeResizeService) { }
 
+  get isContentSearchDialogOpen(): BehaviorSubject<boolean> {
```
Comment (ronnymikalsen) [inline src/lib/src/content-search-dialog/content-search-dialog.service.ts]: You shouldn't expose the subject. Use a Observable instead

File: src/lib/src/content-search-dialog/content-search-dialog.service.ts (hunk 3 above)
Comment (ronnymikalsen) [inline src/lib/src/content-search-dialog/content-search-dialog.service.ts]: Why do you change from boolean to subject?

File: src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { PageService } from '../../page-service/page-service';
+import { Subscription } from 'rxjs/Subscription';
+import { IiifContentSearchService } from '../../iiif-content-search-service/iiif-content-search.service';
```
Comment (ronnymikalsen) [inline src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts]: add a newline before

File: src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { PageService } from '../../page-service/page-service';
```
Comment (ronnymikalsen) [inline src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts]: Should not be together with third party imports

File: src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        } else {
+          const phit = this.searchResult.get(i - 1);
+          return this.searchResult.hits.findIndex(sr => sr.index === phit.index);
+        }
+      }
+    }
+    return this.searchResult.size() - 1;
+  }
+
+  ngOnDestroy() {
```
Comment (ronnymikalsen) [inline src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts]: Where is this function called from?

File: src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  private subscriptions: Array<Subscription> = [];
+
+  constructor(
+    private pageService: PageService,
+    private iiifContentSearchService: IiifContentSearchService
+  ) {
+    this.init();
+  }
+
+  init() {
```
Comment (ronnymikalsen) [inline src/lib/src/core/navigation/content-search-navigation-service/content-search-navigation.service.ts]: Is this method call from anywhere else than the constructor?

File: src/lib/src/content-search-dialog/content-search-dialog.service.ts (hunk 4 above)
Comment (raymondkarstensen) [reply src/lib/src/content-search-dialog/content-search-dialog.service.ts]: You are right. Fixed.

**Pull request merged**